LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -260,20 +286,30 @@ public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaReques
                         }
                     }
 
-                    if (!isConfirmedResubmit)
+                    var sendUpstream = concentratorDeduplicationResult is ConcentratorDeduplicationResult.NotDuplicate || loRaDevice.Deduplication is not DeduplicationMode.Drop;
+
+                    // We send it to the IoT Hub:
+                    // - when it's a new message or it's a resubmission/duplicate but with a strategy that is not drop
+                    // - and it's not a MAC command
+                    if (sendUpstream && payloadPort != LoRaFPort.MacCommand)
                     {
-                        // In case it is a Mac Command only we don't want to send it to the IoT Hub
-                        if (payloadPort != LoRaFPort.MacCommand)
+                        // combine the results of the 2 deduplications: on the concentrator level and on the network server layer
+                        var isDuplicate = concentratorDeduplicationResult is not ConcentratorDeduplicationResult.NotDuplicate || (bundlerResult?.DeduplicationResult?.IsDuplicate ?? false);
+                        if (!await SendDeviceEventAsync(request, loRaDevice, timeWatcher, payloadData, isDuplicate, decryptedPayloadData))
                         {
-                            if (!await SendDeviceEventAsync(request, loRaDevice, timeWatcher, payloadData, bundlerResult?.DeduplicationResult, decryptedPayloadData))
-                            {
-                                // failed to send event to IoT Hub, stop now
-                                return new LoRaDeviceRequestProcessResult(loRaDevice, request, LoRaDeviceRequestFailedReason.IoTHubProblem);
-                            }
+                            // failed to send event to IoT Hub, stop now
+                            return new LoRaDeviceRequestProcessResult(loRaDevice, request, LoRaDeviceRequestFailedReason.IoTHubProblem);
                         }
-
-                        loRaDevice.SetFcntUp(payloadFcntAdjusted);
                     }
+
+                    loRaDevice.SetFcntUp(payloadFcntAdjusted);
+                }
+
+                #region Downstream
+                if (skipDownstreamToAvoidCollisions)
+                {
+                    this.logger.LogInformation($"skipping downstream messages due to deduplication ({timeWatcher.GetElapsedTime()})");
```

Comment:
this should be a debug message. If we log this out, this can cause a lot of messages on devices with a lot of concentrators/devices.

Reply:
Changed it in 76b18e9e0eca82c4d18079629de93bcd795090f7